src/configure

```diff
@@ -1287,6 +1293,28 @@ else
   appropriate configuration for this platform. Please contact the developers."
 fi
 
+if $enable_tcmalloc ; then
+  if [ -d ../tools/gperftools ]; then
+    TCMALLOC_ROOT=$(rel2abs ../tools/gperftools)
+    echo "Checking tcmalloc library in $TCMALLOC_ROOT"
+    if [ -f $TCMALLOC_ROOT/lib/libtcmalloc.so ]; then
+      if [ "$(uname)" == "Linux" ]; then
+        #If profiler is needed, use -ltcmalloc_and_profiler
+        echo "LDLIBS += -Wl,-rpath=$TCMALLOC_ROOT/lib -ltcmalloc" >> kaldi.mk
+      else
+        failure "Now, --enable-tcmalloc option is only supported for Linux"
+      fi
+    else
+      failure "Can't find libtcmalloc.so in ../tools/gperftools,
+               refer to ../tools/extras/install_tcmalloc.sh for instruction."
+    fi
+  else
+    failure "--enable-tcmalloc switched on, but can't find tcmalloc in 
+             ../tools/gperftools, refer to ../tools/extras/install_tcmalloc.sh 
+             for instruction."
+  fi
```

Comment:
I would merge both cases into one. "Can't find tcmalloc in tools/gperftools. Run 'extras/install_tcmalloc.sh' in 'tools/' to install." is good enough for both. install_tcmalloc.sh will print an error if the user tries to do what message says, i.e run `../tools/extras/install_tcmalloc.sh`. "Refer for instructions" is a lie, the script installs stuff without giving any :)

If you split lines, unindent the continuations, it will look weird when printed. `failure` takes any number of arguments and prints them in a single string with a space between, so if a message is concise enough, this will work to split `First half and second half` into two code lines:
```
    failure "First half and" \
            "second half"
```
Just FYI, you may stick `$'\n'` into a string to break lines, outside double quotes, but w/o space, or else Line 2 will start with a space, e.g.: `"Line 1"$'\n'"Line 2"` or `$'Line 1\nLine 2'` do the same thing. You can also do `\'` for `'` inside a `$`-prefixed string.